Build a container image from scratch breaks due to missing libraries

### Describe the bug

Running the instructions here: https://quarkus.io/guides/building-native-image#build-a-container-image-from-scratch
Or with this repo: `https://github.com/andreaTP/quarkus_fully_static/tree/test-quarkus-instructions`
Running this command: `docker build -f Dockerfile.quarkus . -t quarkus_static_quarkus --progress=plain`

The native-image build breaks due to missing libraries.

### Expected behavior

The `native-image` build succeeds.

### Actual behavior

This exception is thrown:
```
#14 14.38 ========================================================================================================================
#14 14.38 GraalVM Native Image: Generating 'code-with-quarkus-1.0.0-SNAPSHOT-runner' (static executable)...
#14 14.38 ========================================================================================================================
#14 14.53 Warning: Feature class io.quarkus.runtime.graal.ResourcesFeature is annotated with the deprecated annotation @AutomaticFeature. Support for this annotation will be removed in a future version of GraalVM. Applications should register a feature using the option --features=io.quarkus.runtime.graal.ResourcesFeature
#14 14.53 Warning: Feature class io.quarkus.runtime.graal.DisableLoggingAutoFeature is annotated with the deprecated annotation @AutomaticFeature. Support for this annotation will be removed in a future version of GraalVM. Applications should register a feature using the option --features=io.quarkus.runtime.graal.DisableLoggingAutoFeature
#14 14.53 Warning: Feature class io.quarkus.runner.AutoFeature is annotated with the deprecated annotation @AutomaticFeature. Support for this annotation will be removed in a future version of GraalVM. Applications should register a feature using the option --features=io.quarkus.runner.AutoFeature
#14 15.48 
#14 15.49 [1/7] Initializing...                                                                                    (0.0s @ 0.10GB)
#14 15.49 Error: Building images for org.graalvm.nativeimage.Platform$LINUX_AMD64 (target libc: musl) requires static JDK libraries.
#14 15.49 Use most recent JDK from https://github.com/graalvm/labs-openjdk-*/releases
#14 15.49 Missing libraries: java, nio, net, zip
#14 15.49 (search path: /opt/mandrel/lib/static/linux-amd64/musl)
#14 15.49 com.oracle.svm.core.util.UserError$UserException: Building images for org.graalvm.nativeimage.Platform$LINUX_AMD64 (target libc: musl) requires static JDK libraries.
#14 15.49 Use most recent JDK from https://github.com/graalvm/labs-openjdk-*/releases
#14 15.49 Missing libraries: java, nio, net, zip
#14 15.49 (search path: /opt/mandrel/lib/static/linux-amd64/musl)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:73)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.guarantee(UserError.java:97)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.NativeLibraries.initCLibraryPath(NativeLibraries.java:350)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.NativeLibraries.<init>(NativeLibraries.java:264)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.setupNativeLibraries(NativeImageGenerator.java:1119)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:915)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:575)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:535)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:580)
#14 15.49       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
#14 15.49 ------------------------------------------------------------------------------------------------------------------------
#14 15.50                          0.2s (5.0% of total time) in 6 GCs | Peak RSS: 0.39GB | CPU load: 3.54
#14 15.50 ========================================================================================================================
#14 15.50 Failed generating 'code-with-quarkus-1.0.0-SNAPSHOT-runner' after 3.1s.
#14 15.57 Error: Image build request failed with exit status 1
#14 15.60 [INFO] ------------------------------------------------------------------------
#14 15.60 [INFO] BUILD FAILURE
#14 15.60 [INFO] ------------------------------------------------------------------------
#14 15.60 [INFO] Total time:  13.782 s
#14 15.60 [INFO] Finished at: 2023-06-21T14:19:00Z
#14 15.60 [INFO] ------------------------------------------------------------------------
#14 15.61 [ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.9.1.Final:build (default) on project code-with-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
#14 15.61 [ERROR]       [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: io.quarkus.deployment.pkg.steps.NativeImageBuildStep$ImageGenerationFailureException: Image generation failed. Exit code: 1
#14 15.61 [ERROR]       at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.imageGenerationFailed(NativeImageBuildStep.java:400)
#14 15.61 [ERROR]       at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:240)
#14 15.61 [ERROR]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
#14 15.61 [ERROR]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
#14 15.61 [ERROR]       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
#14 15.61 [ERROR]       at java.base/java.lang.reflect.Method.invoke(Method.java:568)
#14 15.61 [ERROR]       at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:925)
#14 15.61 [ERROR]       at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
#14 15.61 [ERROR]       at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
#14 15.61 [ERROR]       at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
#14 15.61 [ERROR]       at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
#14 15.61 [ERROR]       at java.base/java.lang.Thread.run(Thread.java:833)
#14 15.61 [ERROR]       at org.jboss.threads.JBossThread.run(JBossThread.java:501)
#14 15.61 [ERROR] -> [Help 1]
#14 15.61 [ERROR] 
#14 15.61 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
#14 15.61 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
#14 15.61 [ERROR] 
#14 15.61 [ERROR] For more information about the errors and possible solutions, please read the following articles:
#14 15.61 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
#14 ERROR: process "/bin/sh -c ./mvnw package -Pnative -DskipTests -Dquarkus.native.additional-build-args=\"--static\",\"--libc=musl\"" did not complete successfully: exit code: 1
------
 > [build 10/10] RUN ./mvnw package -Pnative -DskipTests -Dquarkus.native.additional-build-args="--static","--libc=musl":
#14 15.61 [ERROR]       at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
#14 15.61 [ERROR]       at java.base/java.lang.Thread.run(Thread.java:833)
#14 15.61 [ERROR]       at org.jboss.threads.JBossThread.run(JBossThread.java:501)
#14 15.61 [ERROR] -> [Help 1]
#14 15.61 [ERROR] 
#14 15.61 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
#14 15.61 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
#14 15.61 [ERROR] 
#14 15.61 [ERROR] For more information about the errors and possible solutions, please read the following articles:
#14 15.61 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
------
Dockerfile.quarkus:20
--------------------
  18 |     RUN ./mvnw -B org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline
  19 |     COPY src /code/src
  20 | >>> RUN ./mvnw package -Pnative -DskipTests -Dquarkus.native.additional-build-args="--static","--libc=musl"
  21 |     
  22 |     ## Stage 2 : create the final image
--------------------
ERROR: failed to solve: process "/bin/sh -c ./mvnw package -Pnative -DskipTests -Dquarkus.native.additional-build-args=\"--static\",\"--libc=musl\"" did not complete successfully: exit code: 1
```

please note that the build seems to be fetching `musl` from an unexpected location:

```
(search path: /opt/mandrel/lib/static/linux-amd64/musl)
```

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

OsX

### Output of `java -version`

irrelevant

### GraalVM version (if different from Java)

in the Dockerfile

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

I'll try to dig a bit more, but any additional context around Mandrel and the whole would be really helpful.

--- a/docs/src/main/asciidoc/building-native-image.adoc
+++ b/docs/src/main/asciidoc/building-native-image.adoc
@@ -695,7 +695,7 @@ Sample multistage Dockerfile for building an image from `scratch`:
 [source,dockerfile,subs=attributes+]
 ----
 ## Stage 1 : build with maven builder image with native capabilities
-FROM quay.io/quarkus/ubi-quarkus-mandrel-builder-image:{mandrel-flavor} AS build
+FROM quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:{graalvm-flavor} AS build
 USER root
 RUN microdnf install make gcc
 COPY --chown=quarkus:quarkus mvnw /code/mvnw

